src/compiler.jl

```diff
@@ -101,23 +103,28 @@ automatically and checked for allocations whenever the function changes or when
 types or keyword arguments are provided.
 """
 function compile_callable(f::F, tt::TT=Tuple{}; ignore_throw=true) where {F, TT}
-    # cuda = active_state()
 
     Base.@lock codegen_lock begin
+
         # compile the function
         cache = compiler_cache
         source = GPUCompiler.methodinstance(F, tt)
         rt = Core.Compiler.return_type(f, tt)
 
+        specsig = false # TODO: selectively re-enable after reviewing `uses_specsig` in codegen.cpp
```

Comment:
Here is where we should scan the types and see if there is anything that we cannot correctly translate for specsig.

That might be just checking for `Union`/`UnionAll`? I need to stare at codegen for a while before I'm convinced, though.